lib/gingr/cli.rb

```diff
@@ -60,20 +66,13 @@ def solr(directory)
     option :is_public, type: :boolean, default: true
     def geoserver(filename)
       url = options[:geoserver_url]
-      url ||= if options[:is_public]
-                ENV.fetch('GEOSERVER_URL', Config.default_options[:geoserver_url])
-              else
-                ENV.fetch(
-                  'GEOSERVER_SECURE_URL', Config.default_options[:geoserver_secure_url]
-                )
-              end
-      publisher = GeoserverPublisher.new(url)
-      publisher.update(filename)
-      logger.info("'#{filename}' - published to geoserver #{url} successfully")
+      default = options[:is_public] ? :geoserver_url : :geoserver_secure_url
+      publisher = GeoserverPublisher.new(url, default:)
```

Comment:
I'm not sure "default" is the best name for this since it's more of a "fallback". In any case, I wanted to consolidate the initialization logic around this, and to move it into the class of interest rather than the general purpose Cli class. This is the best I could come up with for now.